packages/react-components/src/components/RichTextEditor/Plugins/UpdateContentPlugin.tsx

```diff
@@ -70,22 +70,17 @@ export class UpdateContentPlugin implements EditorPlugin {
 
       case PluginEventType.ContentChanged:
         if (
-          event.source.toLowerCase() === 'cut' ||
-          (event.source.toLowerCase() === 'keyboard' && (event.data === Keys.BACKSPACE || event.data === Keys.DELETE))
+          event.source === ChangeSource.Cut ||
+          // We need to add the paste source here for an edge case:
+          // when user select an image that's already in the editor, then paste in an image to replace the selected one,
+          // we will only get a paste event.
+          // In this case, we need to update the removedInlineImage array to include the replaced image.
+          event.source === ChangeSource.Paste ||
+          (event.source === ChangeSource.Keyboard && (event.data === Keys.BACKSPACE || event.data === Keys.DELETE))
         ) {
-          imageSrcArray = [];
-          event.contentModel?.blocks.map((block) => {
-            if (block.blockType === 'Paragraph') {
-              const segments = block.segments;
-              segments.map((segment) => {
-                if (segment.segmentType === 'Image') {
-                  imageSrcArray?.push(segment.src);
-                }
-              });
-            }
-          });
+          this.onUpdate(UpdateEvent.ContentChanged, true);
```

Comment:
Let's return this code 
```
event.contentModel?.blocks.map((block) => {
            if (block.blockType === 'Paragraph') {
              const segments = block.segments;
              segments.map((segment) => {
                if (segment.segmentType === 'Image') {
                  imageSrcArray?.push(segment.src);
                }
              });
            }
          });
```
I think it will be better to return array as part of onUpdate function instead of boolean

Reply:
We can move the previous inline images list here as well as they needed only for UpdatePlugin state, not for editor component.

Reply:
> We can move the previous inline images list here as well as they needed only for UpdatePlugin state, not for editor component.

To return an array of removed image src, are you thinking of using the src to compare with the previousInlineImages in the RichTextEditor? If so, using src instead of id as a source of truth is less reliable.

`previousInlineImages` is better placed in the RichTextEditor because we need to set the initial value and clean it in multiple places in the RichTextEditor.

Reply:
Let's leave it as is for now then